src/components/FilterButtonGroup.jsx

```diff
@@ -0,0 +1,107 @@
+
+const StyledToggleButtonGroup = styled(ToggleButtonGroup)(({theme}) => ({
+    '& .MuiToggleButtonGroup-grouped': {
+        margin: theme.spacing(0.5),
+        border: 0,
+        flexWrap: 'wrap',
+        [defaultTheme.breakpoints.down('sm')]: {
+            margin: theme.spacing(0.3),
+            padding: '4px',
+        },
+        '&.Mui-disabled': {
+            border: 0,
+
+        },
+        '&:not(:first-of-type)': {
+            borderRadius: theme.shape.borderRadius,
+        },
+        '&:first-of-type': {
+            borderRadius: theme.shape.borderRadius,
+        },
+    },
+}));
+
+const filterOptions = {
+    levels: [
+        {value: 'basic', label: 'basic'},
+        {value: 'middle', label: 'middle'},
+        {value: 'advanced', label: 'advanced'},
+    ],
+    categories: [
+        {value: 'react', label: 'react'},
+        {value: 'javascript', label: 'javascript'},
+        {value: 'data structure', label: 'data structure'},
+        /*{value: 'html', label: 'html', disabled: true},*/
+    ],
+    labels: [
+        {value: 'frontend', label: 'frontend'},
+        {value: 'backend', label: 'backend'},
+    ],
+};
+
+const ToggleButtonFilterGroup = ({value, onChange, options, ariaLabel}) => {
+    return (
+        <StyledToggleButtonGroup size="small" value={value} onChange={onChange} aria-label={ariaLabel}>
+            {options.map((option) => (
+                <ToggleButton key={option.value} value={option.value} aria-label={option.label}
+                              disabled={option.disabled} sx={{margin:0}}>
+                    {option.label}
+                </ToggleButton>
+            ))}
+        </StyledToggleButtonGroup>
+    );
+};
+
+const FilterButtonGroup = ({changeFilter}) => {
+    const [filters, setFilters] = useState({
+        levels: [],
+        categories: [],
+        labels: [],
+    });
+
+    const handleFilter = (filterType, newValue, filter) => {
+        setFilters((prevFilters) => ({
+            ...prevFilters,
+            [filterType]: newValue,
+        }));
+
+        changeFilter(filterType, filter);
+    };
+
+    const filterTypes = Object.entries(filterOptions);
+
+    return (
+        <Paper id={'filter button group'}
+               elevation={0}
+               sx={{
+                   display: 'flex',
+                   flexWrap: 'wrap',
+                   justifyContent: 'center',
+               }}
+        >
+            {filterTypes.map(([filterType, options], index) => (
+                <React.Fragment key={filterType}>
+                    <ToggleButtonFilterGroup
+                        value={filters[filterType]}
+                        onChange={(e, newValue) => handleFilter(filterType, newValue, e.target.value)}
+                        options={options}
+                        ariaLabel={`filtration by ${filterType}`}
+                    />
+                    {index < filterTypes.length - 1 && (
```

Comment:
The filters work great! I have two suggestions for the UI:
1. Add labels to each filter group, perhaps right above the filter options: 
![image](https://github.com/Code-the-Dream-School/ee-prac-team3-front/assets/6400323/dea259a3-8d25-46fa-b1ba-3903464efe5d)
. Labels help the user get context and also are required for accessibility.
2. Always show the filters and remove the show/hide filter button. When I first landed on the page I didn't know there were filters at all. Since there's so few filters too I don't think it would be a problem to always show them.

What do you think of these suggestions?

Reply:
1. There are already labels for each button separately, as well as for each group. Regarding the labels in the UI, I texted to you in Slack.
2. Done ✅

Reply:
Looks great! Nice work.